subsystems/FPIDSubsystem.java

```
@@ -0,0 +1,140 @@
+
+    /**
+    * the implementing class also needs to get and set encoder values
+    * @param type what type of value we're getting from the encoder. values Position and Velocity as defined above in EncoderValueType
+    */
+    public abstract double getEncoderValue(EncoderValueType type);
+    public abstract void setEncoderPosition(double position);
+
+    /**
+    * the implementing class should decide how to calculate Feedforward 
+    * because we don't know what model to use
+    */
+    public abstract double calcFeedForward(double position, double velocity);
+    public abstract double calcFeedForward();
+
+    /**
+    * it should also do getMeasurement because we don't know what measurement to use
+    * (measurement could be an angle, height, etc)
+    */
+    @Override
+    public abstract double getMeasurement();
+
+    @Override
+    public abstract void periodic();
+
+    /**
+    * move the mechanism based on a given throttle 
+    */
+    public void moveByThrottle(double throttle) {
+        motorGroup.set(throttle * constants.speedMultiplier);
+    }
+
+    /**
+    *  move the mechanism based on a constant multiplier (for operation with buttons)
+    */
+     public void moveByFixedSpeed() {
+        motorGroup.set(constants.speedMultiplier);
+    }
+
+    public void moveFixedReversed() {
+        motorGroup.set(-constants.speedMultiplier);
+    }
+
+    /**
+    * moves the mechanism based on voltage instead of speed
+    */
+    public void moveByVolts(double voltage) {
+        motorGroup.setVoltage(voltage);
+    }
+
+    /**
+    * uses PID output to move the mechanism
+    */
+    @Override
+    protected void useOutput(double output, double setpoint) {
+        try {
+            moveByThrottle(output);
+        }
+        catch(NullPointerException e) {
+            System.out.println("NullPointerException " + e + " from useOutput. \n A constant was likely not given by DriveConstants object");
```

I don't think you need to catch this error, just let it propagate out so people fix it. Otherwise people may not realize there is a problem